buildSrc/src/main/java/org/mozilla/gradle/tasks/ValidateAndroidAppReleaseConfiguration.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.gradle.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import org.mozilla.gradle.GitAggregates
+import org.mozilla.gradle.ext.androidDSLOrThrow
+
+/**
+ * Validates that an Android app is correctly configured for release. When created, this
+ * task will automatically create dependencies upon the release tasks.
+ *
+ * This task currently asserts:
+ * - There are no uncommitted changes in git
+ * - The buildVersion in gradle matches the checked out git tag
+ *
+ * This task can be skipped by supplying the `-Pno-validate` argument.
+ */
+open class ValidateAndroidAppReleaseConfiguration : DefaultTask() {
+    init {
+        group = "Verification"
+        description = "Validates an Android app is correctly configured for release"
+
+        this.onlyIf { !project.hasProperty("no-validate") }
+
+        project.gradle.projectsEvaluated {
+            getAssembleReleaseTasks().forEach {
+                it.dependsOn(this@ValidateAndroidAppReleaseConfiguration)
+            }
+
+            // The assemble release tasks run late during execution so we make
+            getCompileReleaseTasks().forEach {
```

Comment:
I don't understand this comment, what is happening here?

Reply:
It got cut off:
```
// The assemble release tasks run late during execution but the compile tasks run
// early and assemble depends on them so we make our new task run before these compile tasks.
```